cmd/engine/main.go

```diff
@@ -303,36 +348,32 @@ func main() { //nolint:gocyclo
 		cfg.Root = root
 
 		if err := os.MkdirAll(root, 0o700); err != nil {
-			return errors.Wrapf(err, "failed to create %s", root)
+			return fmt.Errorf("failed to create %s: %w", root, err)
 		}
 
 		bklog.G(ctx).Debug("creating engine lockfile")
 		lockPath := filepath.Join(root, "buildkitd.lock")
 		lock := flock.New(lockPath)
 		locked, err := lock.TryLock()
 		if err != nil {
-			return errors.Wrapf(err, "could not lock %s", lockPath)
+			return fmt.Errorf("could not lock %s: %w", lockPath, err)
 		}
 		if !locked {
-			return errors.Errorf("could not lock %s, another instance running?", lockPath)
+			return fmt.Errorf("could not lock %s, another instance running?", lockPath)
 		}
 		defer func() {
 			lock.Unlock()
 			os.RemoveAll(lockPath)
 		}()
 
-		bklog.G(ctx).Debug("creating engine controller")
-		controller, cacheManager, err := newController(ctx, c, &cfg, pubsub)
-		if err != nil {
-			return err
-		}
-		defer controller.Close()
-
-		controller.Register(server)
-
-		go logMetrics(context.Background(), cfg.Root, controller)
-		if cfg.Trace {
-			go logTraceMetrics(context.Background())
+		var traceSocket string
+		if pubsub != nil {
+			bklog.G(ctx).Debug("starting otel controller")
+			traceSocket = filepath.Join(cfg.Root, "otel-grpc.sock")
+			if err := runOtelController(traceSocket, pubsub); err != nil {
+				logrus.Warnf("failed set up otel-grpc controller: %v", err)
+				traceSocket = ""
+			}
```

Comment:
reminder to self: double check this is rm'd after rebase on https://github.com/dagger/dagger/pull/7452#pullrequestreview-2100372653